flink-formats/flink-sql-orc/pom.xml

```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I'm not sure if it is necessary to create new modules just to create the fat jar.
Afaik, you can use the maven shade plugin to create a new jar. I guess the `<finalName>` or `<outputFile>` parameter will be of help for this.

Reply:
If using `<finalName>` or `<outputFile>`, I can not find a way to install or deploy into repository. >_<

Reply:
ah, so if you run "mvn install", the jar won't be added to your local `.m2` directory?

Have you tried these? http://maven.apache.org/plugins/maven-install-plugin/install-file-mojo.html (and maybe also https://maven.apache.org/plugins/maven-deploy-plugin/deploy-file-mojo.html ?)

Reply:
My idea is to automatically upload to Maven repositories, through which users can download by themselves. (The previous jar-with-dependency didn't work too)
Like jars in https://repo.maven.apache.org/maven2/org/apache/flink/flink-sql-connector-kafka_2.11/1.10.1/

Reply:
I found a JIRA related to sql-jars: https://issues.apache.org/jira/browse/FLINK-11026

Reply:
PR descriptor updated.
Profile with maven-shade-plugin and `shadedClassifierName` can create a bundled jar and upload jar to maven repositories.
But for accurate notice files it is necessary to be able to supply each artifact with distinct NOTICE files, we need create a separate module.

Reply:
> But for accurate notice files it is necessary to be able to supply each artifact with distinct NOTICE files, we need create a separate module.

But we would only need one NOTICE file in `flink-orc` which we will use for the shaded fat jar. For the regular `flink-orc.jar` we don't need a NOTICE file, as it contains only Flink binaries.

Sorry that I'm making this so complicated. I'm just trying to make sure that we are not introducing unnecessary complexity here, and that the NOTICE files are valid.

Reply:
> Sorry that I'm making this so complicated.

It's all right, your review makes things clearer~

I thought the meaning of https://github.com/apache/flink/pull/7251 is:
For the regular `flink-orc.jar` should has a `NOTICE` file too.

If we create a `flink-sql-orc` in `flink-orc` module with classifier `sql-jar`. Then, there are two jars in artifact `flink-orc`.
And they should have different `NOTICE` files (Which is hard to do). One has many dependencies, but another just is an simple file like:
```
// ------------------------------------------------------------------
// NOTICE file corresponding to the section 4d of The Apache License,
// Version 2.0, in this case for Apache Flink
// ------------------------------------------------------------------

Apache Flink
Copyright 2006-2020 The Apache Software Foundation

This product includes software developed at
The Apache Software Foundation (http://www.apache.org/).

flink-orc
Copyright 2014-2020 The Apache Software Foundation
```

Correct me if I wrong.
CC: @zentol

Reply:
Thanks a lot for the explanation and reference. It seems that adding a second module is the way to go!